### DISCUSSION TITLE
Toolbar with button to change a property value...

### DISCUSSION THREAD

**Discussion by melmaninga** [chrisgurney/obsidian-note-toolbar#206] [Scripting]
I am beginning to explore Note Toolbar.  My use case would be to have a toolbar set that would give the user the ability to change the status property (YAML) of the current note by clicking on a button.  I would imagine you would give different types of notes different types of status choices by using the notetoolbar property.

Is this possible?

**Comment by chrisgurney** [comment]
@melmaninga Note Toolbar can't change properties directly, but if you use Templater, Dataview, or JS Engine, you can add an item with one of those three types and execute a script.

With **Templater**, the [`SwapToolbars` script](https://raw.githubusercontent.com/chrisgurney/obsidian-note-toolbar/refs/heads/master/examples/Scripts/Templater/SwapToolbars.js) changes the `notetoolbar` property when the user selects from a list of available toolbars. If you're looking for an example of how to change properties (and you use Templater), you might be able adapt it to your use case from there.

If you use **Dataview** or JavaScript (via **JS Engine**), I would be happy to try to find an example to add to the repo that does that.

LMK if this helps!

**Reply by chrisgurney** [reply]
Is this you? https://www.reddit.com/r/ObsidianMD/comments/1hpgk6s/which_plugin_would_you_use_to_dynamically_change/

If so, and you're open to using **Templater**, I think that `SwapToolbars` script could definitely be adapted to your use case. I'll take a look.

~~I'm not sure how to go about making a Suggester box appear with Dataview or JS Engine to select from.~~ You can show a Suggester with **Dataview** or **JS Engine** using the Note Toolbar API, if you would rather use one of those options.

**Reply by chrisgurney** [reply]
Here's a working example using **Templater**. Hope this helps!

```javascript
/**
 * Updates the note's status property based on a hard-coded list of statuses.
 *
 * Usage:
 * - In the Templater plugin's settings, add the folder this script is in under "User script functions".
 * - Add a Templater item to your toolbar and select "Execute Templater command".
 * - In the "Templater command" field: tp.user.StatusSelector(tp, "colour")
 */

module.exports = async function StatusSelector(tp, statusCategory) {

    const statusOptions = {
        colour: ['red', 'orange', 'yellow', 'green', 'blue', 'indigo', 'violet'],
        vegetable: ['root vegetables', 'cruciferous vegetables', 'greens', 'nightshades'],
    };

    const validStatusOptions = statusOptions[statusCategory];

    if (validStatusOptions) {

        // prompt the user to pick a status
        const selectedStatus = await tp.system.suggester((item) => item, validStatusOptions, true, "Select a status:");

        // get the current file
        const currentFile = app.workspace.getActiveFile();
        await app.fileManager.processFrontMatter(currentFile, async (frontmatter) => {
            // update the status with the selected value
            frontmatter["status"] = selectedStatus;
        });
        
    }
    else {

        new Notice("Invalid status category: " + statusCategory);

    }

}
```

**Reply by 19jmrs** [reply]
@chrisgurney I have this script
``` js
module.exports = async function updateMain() {
  let subProject = "";
  const currentFile = app.workspace.getActiveFile();
  const pathArray = currentFile.path.split("/");

  if (pathArray.length > 1) {
    // -1 will be the markdown file itself
    subProject = pathArray.at(-2);
  }

  await app.fileManager.processFrontMatter(currentFile, (frontmatter) => {
    frontmatter["Main"] = subProject;
  });
};
```

And after this I simply call in the template : <% tp.user.updateMain() %> but for some reason it does not work, but when I run it in the console it works fine, any idea of the cause of the bug?

**Reply by chrisgurney** [reply]
@19jmrs Try using this configuration for your toolbar item, and let me know if this works for you:
![CleanShot 2025-01-22 at 14 56 50@2x](https://github.com/user-attachments/assets/44d936fd-c683-4469-8dd7-064f955afe7e)

**Comment by melmaninga** [comment]
Thanks Chris!

Back on it in about a week.

---
Aloha!

Melvin Wilson Jr.
(Sent while mobile)

**Comment by chrisgurney** [comment]
Update: This is even easier to do now, using the [`ntb.setProperty()` API](https://github.com/chrisgurney/obsidian-note-toolbar/wiki/Note-Toolbar-API#setproperty).
